beep/protocol/biologic_mb_to_maccor.py

```diff
@@ -0,0 +1,404 @@
+        elif unit == "mA":
+            return "{}E-3".format(str_val)
+        elif unit in ("\N{Greek Small Letter Mu}A", "\N{Micro Sign}A"):
+            return "{}E-6".format(str_val)
+        elif unit == "nA":
+            return "{}E-9".format(str_val)
+        elif unit == "pA":
+            return "{}E-12".format(str_val)
+        else:
+            raise Exception(
+                "Unsupported current unit: {} for {}, at seq {}".format(
+                    unit, field, seq_num
+                )
+            )
+
+    @staticmethod
+    def _convert_voltage(str_val, unit, field, seq_num):
+        if unit == "V":
+            return "{}".format(str_val)
+        elif unit == "mV":
+            return "{}E-3".format(str_val)
+        else:
+            raise Exception(
+                "Unsupported voltage unit: {} for {}, at seq {}".format(
+                    unit, field, seq_num
+                )
+            )
+
+    @staticmethod
+    def _convert_resistance(str_val, unit, field, seq_num):
+        if unit == "MOhm":
+            return "{}E6".format(str_val)
+        elif unit == "kOhm":
+            return "{}E3".format(str_val)
+        elif unit == "Ohm":
+            return "{}".format(str_val)
+        elif unit == "mOhm":
+            return "{}E-3".format(str_val)
+        elif unit in ("\N{Greek Small Letter Mu}Ohm", "\N{Micro Sign}Ohm"):
+            return "{}E-6".format(str_val)
+        else:
+            raise Exception(
+                "Unsupported resistance unit: {} for {}, at seq {}".format(
+                    unit, field, seq_num
+                )
+            )
+
+    @staticmethod
+    def _convert_power(str_val, unit, field, seq_num):
+        if unit == "W":
+            return "{}".format(str_val)
+        elif unit == "mW":
+            return "{}E-3".format(str_val)
+        elif unit in ("\N{Greek Small Letter Mu}W", "\N{Micro Sign}W"):
+            return "{}E-6".format(str_val)
+        else:
+            raise Exception(
+                "Unsupported power unit: {} for {}, at seq {}".format(
+                    unit, field, seq_num
+                )
+            )
+
+    @classmethod
+    def _create_loop_step(cls, seq, seq_num_by_step_num):
+        step = deepcopy(_blank_step)
+        # spacing may be important
+        step["StepType"] = " Loop 1 "
+        # unclear if spacing is necessary
+        step["StepMode"] = "        "
+
+        loop_end_entry = deepcopy(_blank_end_entry)
+
+        loop_count = int(seq["ctrl_repeat"])
+        loop_to_seq = int(seq["ctrl_seq"])
+        loop_to_step = seq_num_by_step_num[loop_to_seq]
+
+        loop_end_entry["EndType"] = "Loop Cnt"
+        loop_end_entry["Oper"] = " = "
+        loop_end_entry["StepValue"] = "{}".format(loop_count)
+        loop_end_entry["Step"] = cls._stringify_step_num(loop_to_step)
+        assert len(loop_end_entry["Step"]) == 3
+
+        step["Ends"] = OrderedDict({"EndEntry": loop_end_entry,})
```

Comment:
Ideally black would fix this, but here the issue is the trailing comma.  I'd just remove it manually for now.

Reply:
See [this issue](https://github.com/psf/black/issues/1288).